app/channel.go

```diff
@@ -2585,66 +2585,6 @@ func (a *App) MarkChannelAsUnreadFromPost(postID string, userID string, collapse
-		if post.RootId == "" {
-			threadId = post.Id
-		}
-
-		var nfErr *store.ErrNotFound
-		threadMembership, storeErr := a.Srv().Store.Thread().GetMembershipForUser(user.Id, threadId)
-		if storeErr != nil && !errors.As(storeErr, &nfErr) {
-			return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, storeErr.Error(), http.StatusInternalServerError)
-		}
-		// if this post was not followed before, create thread membership and update mention count
-		if threadMembership == nil {
-			opts := store.ThreadMembershipOpts{
-				Following:             false,
-				IncrementMentions:     false,
-				UpdateFollowing:       true,
-				UpdateViewedTimestamp: true,
-				UpdateParticipants:    false,
-			}
-			threadMembership, storeErr = a.Srv().Store.Thread().MaintainMembership(user.Id, threadId, opts)
-			if storeErr != nil && !errors.As(storeErr, &nfErr) {
-				return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, storeErr.Error(), http.StatusInternalServerError)
-			}
-			threadData, storeErr2 := a.Srv().Store.Thread().Get(threadId)
-			if storeErr2 != nil {
-				return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, storeErr2.Error(), http.StatusInternalServerError)
-			}
-			if threadData != nil && threadMembership != nil && threadMembership.Following {
-				channel, nErr := a.Srv().Store.Channel().Get(post.ChannelId, true)
-				if nErr != nil {
-					return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, nErr.Error(), http.StatusInternalServerError)
-				}
-				threadMembership.UnreadMentions, err = a.countThreadMentions(user, post, channel.TeamId, post.UpdateAt-1)
-				if err != nil {
-					return nil, err
-				}
-				_, nErr = a.Srv().Store.Thread().UpdateMembership(threadMembership)
-				if nErr != nil {
-					return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, nErr.Error(), http.StatusInternalServerError)
-				}
-				thread, nErr := a.Srv().Store.Thread().GetThreadForUser(channel.TeamId, threadMembership, true)
-				if nErr != nil {
-					return nil, model.NewAppError("MarkChannelAsUnreadFromPost", "app.channel.update_last_viewed_at_post.app_error", nil, nErr.Error(), http.StatusInternalServerError)
-				}
-				a.sanitizeProfiles(thread.Participants, false)
-				thread.Post.SanitizeProps()
-				payload, jsonErr := json.Marshal(thread)
-				if jsonErr != nil {
-					mlog.Warn("Failed to encode thread to JSON")
-				}
-				message := model.NewWebSocketEvent(model.WebsocketEventThreadUpdated, channel.TeamId, "", userID, nil)
-				message.Add("thread", string(payload))
-				a.Publish(message)
-			}
-		}
-	}
-
 	channelUnread, nErr := a.Srv().Store.Channel().UpdateLastViewedAtPost(post, userID, unreadMentions, unreadMentionsRoot, true)
```

Comment:
Discussed this with @koox00 on zoom

This is all dead code. If a threadMembership did not exist, we made one with `Following` set to false. Because `Following` was set to false, the condition on line 2618 was never true, and therefore counts and mentions are never updated.

Doesn't seem to be a need to create a ThreadMembership here, especially since `Following` is always false. 

I've removed all deprecated tests as well.